New doc "Migrate to Runtime" + more Runtime docs (#874)
astronomer/docs: software/migrate-to-runtime.md

---
title: 'Migrate a Deployment from Astronomer Certified to Astro Runtime'
sidebar_label: 'Migrate to Astro Runtime'
id: migrate-to-runtime
description: Run an upgrade progress to migrate your Software Deployment from Astronomer Certified to Astro Runtime.
---

> Astro Runtime builds on the reliability of Astronomer Certified with new features and provider packages. Astronomer plans to no longer release new versions of Astronomer Certified starting with a minor release of Airflow to be determined. To avoid disruption in the future, Astronomer recommends migrating Deployments from Certified to Runtime now.

Reviewer: ```suggestion
Astro Runtime builds on the reliability of Astronomer Certified and adds new features and provider packages. Astronomer will soon stop releasing new versions of Astronomer Certified. To avoid future service disruptions, Astronomer recommends migrating your Deployments from Astronomer Certified to Astro Runtime now.
```

Migrating a Deployment from Astronomer Certified is similar to the standard upgrade process. There are no known disruptions when migrating a Deployment from Astronomer Certified to the equivalent version of Astro Runtime.

## Differences between Astro Runtime and Astronomer Certified

Functionally, Runtime images are similar to Certified images. They both include:

- Same-day support for Apache Airflow releases.
- Extended support lifecycles.
- Regularly backported bug and security fixes.

Astro Runtime includes additional features which are not available in Astronomer Certified images, including:

> - The `astronomer-providers` package, which includes a set of deferrable operators build and maintained by Astronomer.

Reviewer: ```suggestion
- The `astronomer-providers` package, which includes a set of deferrable operators built and maintained by Astronomer.
```

> - Airflow UI improvements, such as the showing your Deployment's Docker image tag in the footer of all UI pages.

Reviewer: ```suggestion
- Airflow UI improvements, such as showing the Deployment Docker image tag in the footer of all UI pages.
```

> - Future Runtime-exclusive features, such as new Airflow components and improvements to the DAG development experience.

Reviewer: ```suggestion
- Future Astro Runtime-exclusive features, such as new Airflow components and improvements to the DAG development experience.
```
Question: Should we be referring to it as _Astro Runtime_ or can it be truncated to _Runtime_? In my opinion, we should always use _Astro Runtime_ as it reenforces ownership and branding.

See [Runtime Architecture](runtime-architecture.md) for more detailed information about Runtime's distribution and features.

All versions of Astronomer Certified have an equivalent version of Astro Runtime. To see the equivalent version of Astro Runtime for a Deployment running Astronomer Certified, open the Deployment in the Software UI and go to **Settings**. The equivalent version of Astro Runtime is shown in the **Migrate to Runtime-[Version number]** button.

## Step 1: Start the migration process

1. In the Software UI, open your Deployment.

> 2. In the **Settings** tab, click **Migrate to Runtime-[Version Number]**.

Reviewer: ```suggestion
2. Click the **Settings** tab and then click **Migrate to Runtime-[Version Number]**.
```
From the style guide "..use _click_ for most targets, such as buttons, links, list items, and radio buttons".

See [Runtime Architecture](runtime-image-architecture.md) for more detailed information about Runtime's distribution and features.

All versions of Astronomer Certified have an equivalent version of Astro Runtime. To see the equivalent version of Astro Runtime for a Deployment running Astronomer Certified, open the Deployment in the Software UI and go to **Settings**. The equivalent version of Astro Runtime is shown in the **Migrate to Runtime-[Version number]** button.

## Step 1: Start the migration process

1. In the Software UI, open your Deployment.
2. In the **Settings** tab, click **Migrate to Runtime-[Version Number]**.

You can't simultaneously migrate to Astro Runtime and upgrade your Deployment. To upgrade to a later version of Runtime than your equivalent Astronomer Certified version, you must first upgrade Astronomer Certified before migrating to Runtime. For example, the upgrade path from Astronomer Certified 2.2.3 (Airflow 2.2.3) to Runtime 5.0.4 (Airflow 2.3.0) would be:

Astronomer Certified 2.2.3 > Astronomer Certified 2.3.0 > Astro Runtime 5.0.4

> If you prefer to use the Astro CLI, you can run `astro deployment runtime migrate --deployment-id=<your-deployment-id>` to start the upgrade process.

Reviewer 2: Have this be a tip?

Author: Rob has avoided using tips for alternative CLI setups in other docs, so I was following that standard here.

## Step 2: Migrate your Astro project

1. In your Astro project, open your `Dockerfile`.

> 2. Update the `FROM` line of your project's `Dockerfile` to reference a new Astronomer image. For example, to upgrade to the latest version of Astro Runtime, you would change the `FROM` line to:

Reviewer 2: ```suggestion
2. Update the `FROM` line of your project's `Dockerfile` to reference a new Astro Runtime image. For example, to migrate to Astro Runtime 5.0.4, you would change the `FROM` line to:
```

    <pre><code parentName="pre">{`FROM quay.io/astronomer/astro-runtime:${siteVariables.runtimeVersion}`}</code></pre>

>     For a list of currently supported Astronomer images, see:

Reviewer 2: ```suggestion
    For a list of currently supported Astro Runtime images, see:
```

Should this just be Runtime? Or is the info here moreso the Runtime/Certified and Airflow parity concern?

Author: We link to both below. Given that we're providing _an_ option to both upgrade and migrate, I think we should include both image sets here.

    - [Astronomer Certified Lifecycle Schedule](ac-support-policy.md#astronomer-certified-lifecycle-schedule)
    - [Astro Runtime Lifecycle Schedule](https://docs.astronomer.io/astro/runtime-version-lifecycle-policy#astro-runtime-lifecycle-schedule)

  :::warning

>   After you upgrade your Airflow version, you can't revert to an earlier version.

Reviewer 2: If you agree with my earlier comment, add here too

  :::

> 3. Optional. Test your upgrade on your local machine by running:

Reviewer 2: ```suggestion
3. Optional. Test your migration to Astro Runtime locally by running:
```

    ```sh
    astro dev restart
    ```

    All 4 running Docker containers for each of the Airflow components (webserver, scheduler, Postgres, triggerer) restart and begin running your new image.

>     Open the Airflow UI at `localhost:8080` and go to **About** > **Version**. Confirm that you're running the correct Airflow version.

Reviewer 2: Same here

## Step 3: Deploy to Astronomer

> 1. To push your upgrade to a Deployment on Astronomer Software, run:

Reviewer 2: ```suggestion
1. To push your migration from Astronomer Certified to Astro Runtime on Astronomer Software, run:
```

---
title: 'Migrate a Deployment from Astronomer Certified to Astro Runtime'
sidebar_label: 'Migrate to Astro Runtime'
id: migrate-to-runtime
description: Run an upgrade progress to migrate your Software Deployment from Astronomer Certified (AC) to Astro Runtime.
---

import {siteVariables} from '@site/src/versions';

Astro Runtime builds on the reliability of Astronomer Certified (AC) with new features that center on usability and performance.

Astronomer will no longer release new versions of AC starting with the release of Apache Airflow 2.4. To avoid disruption in the future, Astronomer recommends migrating Deployments from AC to Astro Runtime as soon as your team is ready.

Migrating a Deployment to Astro Runtime is similar to the standard upgrade process. There are no known disruptions when migrating a Deployment from AC to the equivalent version of Astro Runtime.

> ## Differences between Astro Runtime and AC

Reviewer 2: ```suggestion
## Differences between Astro Runtime and Astronomer Certified
```

Functionally, Runtime images are similar to Certified images. They both include:

- Timely support for new patch, minor, and major versions of Apache Airflow.
- Support lifecycles that extend beyond those offered by the open source community.
- Regularly backported bug and security fixes.

Astro Runtime includes additional features which are not available in AC images, including:

> - The `astronomer-providers` package, which is an open source collection of Apache Airflow providers and modules maintained by Astronomer.
> - Airflow UI improvements, such as showing the Deployment Docker image tag in the footer of all UI pages.
> - Future Astro Runtime-exclusive features, such as new Airflow components and improvements to the DAG development experience.

Reviewer 2: ```suggestion
- Exclusive features to enrich the task execution experience, including smart task concurrency defaults and high availability configurations.
- The `astronomer-providers` package, which is an open source collection of Apache Airflow providers and modules maintained by Astronomer.
- Airflow UI improvements, such as showing the Deployment Docker image tag in the footer of all UI pages.
```

Tweaked section here to add the task execution stuff and removed the future-facing statement. Don't feel too strongly if you wanna add that last bullet point back!

See [Runtime Architecture](runtime-image-architecture.md) for more detailed information about Runtime's distribution and features.

All versions of AC have an equivalent version of Astro Runtime. To see the equivalent version of Astro Runtime for a Deployment running AC, open the Deployment in the Software UI and go to **Settings**. The equivalent version of Astro Runtime is shown in the **Migrate to Runtime-[Version number]** button.

## Step 1: Start the migration process

1. In the Software UI, open your Deployment.
2. Click the **Settings** tab and then click **Migrate to Runtime-[Version Number]**.

You can't simultaneously migrate to Astro Runtime and upgrade your Deployment. To upgrade to a later version of Runtime than your equivalent AC version, you must first either upgrade Astronomer Certified or migrate to Astro Runtime before upgrading. For example, the upgrade path from AC 2.2.3 (Airflow 2.2.3) to Runtime 5.0.4 (Airflow 2.3.0) would be:

AC 2.2.3 > AC 2.3.0 > Astro Runtime 5.0.4

If you prefer to use the Astro CLI, you can run `astro deployment runtime migrate --deployment-id=<your-deployment-id>` to start the upgrade process.

> ## Step 2: Migrate your Astro project

Reviewer 2: ```suggestion
## Step 2: Modify your Astro project
```

I don't feel like you're migrating the project itself, you're just migrating to Runtime from AC -- will let you tweak wording, but maybe something closer to what I suggested?

Author: The step before is "Start the migration", whereas this is the real bit of migration in the whole setup. I'd prefer to have a step where people can go "oh, that's where I'm actually doing the migration"

1. In your Astro project, open your `Dockerfile`.
2. Update the `FROM` line of your project's `Dockerfile` to reference a new Astro Runtime image. For example, to migrate to Astro Runtime 5.0.4, you would change the `FROM` line to:

    ```sh
    FROM quay.io/astronomer/astro-runtime:5.0.4
    ```

    For a list of currently supported Astronomer images, see:

    - [AC Lifecycle Schedule](ac-support-policy.md#astronomer-certified-lifecycle-schedule)
    - [Astro Runtime Lifecycle Schedule](https://docs.astronomer.io/astro/runtime-version-lifecycle-policy#astro-runtime-lifecycle-schedule)

  :::warning

  Astronomer does not support Airflow downgrades. After you upgrade your Airflow version, you can't revert to an earlier version.

  :::

3. Optional. Test your migration to Astro Runtime locally by running:

    ```sh
    astro dev restart
    ```

    All 4 running Docker containers for each of the Airflow components restart and begin running your new image.

>     To confirm that your migration was successful, open the Airflow UI at `localhost:8080` and go to **About** > **Version**. This page should list your new Astro Runtime version.

Reviewer 2: WHOOPS I don't think this is true -- I think you confirm in this footer:

<img width="437" alt="image" src="https://user-images.githubusercontent.com/31111752/178076786-8cf4d523-bbe8-4ad5-8548-88d6c2db04e3.png">

Should be what's here: https://docs.astronomer.io/astro/upgrade-runtime#step-4-confirm-your-upgrade-on-astro

Reviewer 2: If you change this make sure you change other instances

## Step 3: Deploy to Astronomer

1. To push your migration from Astronomer Certified to Astro Runtime on Astronomer Software, run:

    ```sh
    astro deploy
    ```

2. In the Software UI, open your Deployment and click **Open Airflow**.

> 3. Go to **About** > **Version**. Confirm that you're running the correct version of Astro Runtime.

Reviewer 2: Same here I think whoops
